internal/provider/provider.go

```diff
@@ -832,6 +832,9 @@ func buildOidcCredential(model providerData, options azidentity.DefaultAzureCred
 	if err != nil {
 		return nil, err
 	}
+	if model.OIDCToken.ValueString() == "" && model.GetOIDCTokenFilePath() == "" && model.OIDCRequestToken.ValueString() == "" && model.OIDCRequestURL.ValueString() == "" {
```

Comment:
should be `||`, not `&&`?

Reply:
No, OIDC doesn't require all of them to be set.